### PR TITLE
feat: support expanding config value, convert json to yaml

### PR DESCRIPTION
**Pull request by debendraoli**
## Description:

This change allows expanding config values with OS Environment variables eliminating hard coded values on config file.

### Commit Message

```bash
feat: support expanding config value, convert json to yaml
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

N/A.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the unit tests
- [x] I only have one commit (if not, squash them into one commit).
- [x] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: test/e2e/config.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  keystore_file: "$BASE_PATH/contracts/javascore/test-lib/conf/godWallet.json"
+  keystore_password: gochain
+  chain_config:
+    type: icon
+    name: goloop2
+    chain_id: icon2
+    bech32_prefix: hx
+    bin: "goloop"
+    image:
+      repository: debendraoli/relayer
```
Comment (AntonAndell) [inline test/e2e/config.yaml:39]: This should be the goloop image for now, relay should be a separate config or hard coded in the tests

**Review by AntonAndell — CHANGES_REQUESTED** [review]

File: test/e2e/icon-icon_test.go [hunk 2]
```diff
@@ -32,7 +35,7 @@ func TestICONToICON(t *testing.T) {
 	if err != nil {
 		return
 	}
-	optionDocker := relayer.CustomDockerImage("relayer", "latest", "100:1000")
+	optionDocker := relayer.CustomDockerImage(cfg.Counterparty.ChainConfig.Images.Repository, cfg.Counterparty.ChainConfig.Images.Version, "100:1000")
```
[the remark this answers is not in the document]
Comment (debendraoli) [reply test/e2e/icon-icon_test.go:38]: Sure. Wouldn't it is better taking values from config rather hard coded value?

[the remark this answers is not in the document]
Comment (AntonAndell) [reply test/e2e/icon-icon_test.go:38]: Yeah taking from config would be fine but will be the same for all tests so a constant somewhere would also be okey

[the remark this answers is not in the document]
Comment (redlarva) [reply test/e2e/icon-icon_test.go:38]: Yeah I agree with @AntonAndell ,We can use constant as we will be using same/latest version of relayer most of the time.I will revert this to use constant for relayer

[the remark this answers is not in the document]
Comment (debendraoli) [reply test/e2e/icon-icon_test.go]: I have made use of constant and also overriding option when required from env.

File: test/e2e/config.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  keystore_file: "$BASE_PATH/contracts/javascore/test-lib/conf/godWallet.json"
+  keystore_password: gochain
+  chain_config:
+    type: icon
+    name: goloop2
+    chain_id: icon2
+    bech32_prefix: hx
+    bin: "$GOPATH/bin/goloop"
+    image:
+      repository: iconloop/icon2-node
```
Comment (AntonAndell) [inline test/e2e/config.yaml:39]: What is this docker? and why is it not the same as the other ICON chain?

**Review by AntonAndell — COMMENTED** [review]
Looks good just one question

**Review by AntonAndell — APPROVED** [review]

**Review by redlarva — APPROVED** [review]
looks good

**Pull request merged**